service/http/dffml_service_http/cli.py

```diff
@@ -129,78 +154,75 @@ class CreateTLS(TLSCMD):
     client = CreateTLSClient
 
 
-class MultiCommCMD(CMD):
-
-    arg_mc_config = Arg(
-        "-mc-config",
-        dest="mc_config",
-        default=None,
-        help="MultiComm config directory",
+@config
+class MultiCommCMDConfig:
+    mc_config: str = field(
+        "MultiComm config directory", default=None,
     )
-    arg_mc_atomic = Arg(
-        "-mc-atomic",
-        dest="mc_atomic",
+    mc_atomic: bool = field(
+        "No routes other than dataflows registered at startup",
         action="store_true",
         default=False,
-        help="No routes other than dataflows registered at startup",
     )
 
 
-class Server(TLSCMD, MultiCommCMD, Routes):
-    """
-    HTTP server providing access to DFFML APIs
-    """
+class MultiCommCMD(CMD):
+
+    CONFIG = MultiCommCMDConfig
 
-    # Used for testing
-    RUN_YIELD_START = False
-    RUN_YIELD_FINISH = False
-    INSECURE_NO_TLS = False
 
-    arg_port = Arg("-port", help="Port to bind to", type=int, default=8080)
-    arg_addr = Arg("-addr", help="Address to bind to", default="127.0.0.1")
-    arg_upload_dir = Arg(
-        "-upload-dir",
-        help="Directory to store uploaded files in",
-        default=None,
+@config
+class ServerConfig(TLSCMDConfig, MultiCommCMDConfig):
+    port: int = field(
+        "Port to bind to", default=8080,
+    )
+    addr: str = field(
+        "Address to bind to", default="127.0.0.1",
     )
-    arg_static = Arg(
-        "-static", help="Directory to serve static content from", default=None
+    upload_dir: str = field(
+        "Directory to store uploaded files in", default=None,
     )
-    arg_js = Arg(
-        "-js",
-        help="Serve JavaScript API file at /api.js",
+    static: str = field(
+        "Directory to serve static content from", default=None,
+    )
+    js: bool = field(
+        "Serve JavaScript API file at /api.js",
         default=False,
         action="store_true",
     )
-    arg_insecure = Arg(
-        "-insecure",
-        help="Start without TLS encryption",
-        action="store_true",
-        default=False,
+    insecure: bool = field(
+        "Start without TLS encryption", action="store_true", default=False,
     )
-    arg_cors_domains = Arg(
-        "-cors-domains",
-        help="Domains to allow CORS for (see keys in defaults dict for aiohttp_cors.setup)",
-        nargs="+",
-        default=[],
+    cors_domains: List[str] = field(
```

Comment:
The service/http tests are raising `TypeError: '_MISSING_TYPE' object is not iterable` for `cors_domain`. not sure what's happening here. @pdxjohnny Can you provide some input?

Reply:
One of the several places it's raising this: 
```console
======================================================================
ERROR: test_update (tests.test_routes.TestRoutesSource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tron/Desktop/dffml/dffml/util/asynctestcase.py", line 68, in run_it
    result = self.loop.run_until_complete(coro(*args, **kwargs))
  File "/home/tron/anaconda3/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/home/tron/Desktop/dffml/service/http/tests/test_routes.py", line 349, in setUp
    await super().setUp()
  File "/home/tron/Desktop/dffml/service/http/dffml_service_http/util/testing.py", line 114, in setUp
    await self.tserver.start(self.cli.run())
  File "/home/tron/Desktop/dffml/service/http/dffml_service_http/util/testing.py", line 84, in start
    raise exception
  File "/home/tron/Desktop/dffml/service/http/dffml_service_http/cli.py", line 251, in run
    await self.setup()
  File "/home/tron/Desktop/dffml/service/http/dffml_service_http/routes.py", line 752, in setup
    for domain in self.cors_domains
TypeError: '_MISSING_TYPE' object is not iterable

```